storage/client_split_test.go

```diff
@@ -459,3 +460,83 @@ func TestStoreRangeSplitOnConfigs(t *testing.T) {
 		t.Errorf("expected splits not found: %s", err)
 	}
 }
+
+// TestStoreRangeManySplits splits many ranges at once.
+func TestStoreRangeManySplits(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	store, stopper := createTestStore(t)
+	defer stopper.Stop()
+
+	// Write zone configs to trigger the first round of splits.
+	zoneConfig := &proto.ZoneConfig{}
+	b := &client.Batch{}
+	for i := 0; i < 20; i++ {
+		key := proto.Key(fmt.Sprintf("db%02d", 20-i))
+		b.Put(keys.MakeKey(keys.ConfigZonePrefix, key), zoneConfig)
+	}
+	if err := store.DB().Run(b); err != nil {
+		t.Fatal(err)
+	}
+
+	// Check that we finish splitting in allotted time.
+	expKeys := []proto.Key{}
+	for i := 1; i < 22; i++ {
```

Comment:
this `22` is still a magic number - perhaps extract `20` above into a variable and then explain why this is 2 more than that value?